src/wallet/wallet.h

```diff
@@ -475,7 +475,38 @@ class CWalletTx : public CMerkleTx
 };
 
 
-using CInputCoin = std::pair<const CWalletTx*, unsigned int>;
+class CInputCoin {
+public:
+    CInputCoin()

```

Comment:
I don't think using `optional` would make the code more clear. I never saw it used anywhere in bitcoin.
What about using `CInputCoin*` for coinLowestLarger instead?